### PR TITLE
Completes build config update

### PR DESCRIPTION
**Pull request by seia-soto**
- update build config for all packages
- drop "declarationDir" from tsconfig which will cause conflict with @rollup/plugin-typescript
- properly specify file location with prefix of "./"
- fix up some file orders
- fix the declaration directory to "./dist/types"
- create npmignore unless exists

refs https://github.com/rollup/plugins/issues/1230

### CODE REVIEW

**Comment by seia-soto** [comment]
Leaving a reference to upstream: https://github.com/ghostery/adblocker/pull/4061

**Comment by seia-soto** [comment]
<details>
<summary>Details</summary>

```
% tree -f -L 4 . | grep -i dist
│   ├── ./adblocker/dist
│   │   ├── ./adblocker/dist/adblocker.cjs
│   │   ├── ./adblocker/dist/adblocker.cjs.map
│   │   ├── ./adblocker/dist/adblocker.d.ts
│   │   ├── ./adblocker/dist/adblocker.js
│   │   ├── ./adblocker/dist/adblocker.js.map
│   │   ├── ./adblocker/dist/adblocker.umd.min.js
│   │   ├── ./adblocker/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker/dist/src
│   │   │   ├── ./adblocker/dist/src/codebooks
│   │   │   ├── ./adblocker/dist/src/compact-set.cjs
│   │   │   ├── ./adblocker/dist/src/compact-set.cjs.map
│   │   │   ├── ./adblocker/dist/src/compact-set.d.ts
│   │   │   ├── ./adblocker/dist/src/compact-set.js
│   │   │   ├── ./adblocker/dist/src/compact-set.js.map
│   │   │   ├── ./adblocker/dist/src/compression.cjs
│   │   │   ├── ./adblocker/dist/src/compression.cjs.map
│   │   │   ├── ./adblocker/dist/src/compression.d.ts
│   │   │   ├── ./adblocker/dist/src/compression.js
│   │   │   ├── ./adblocker/dist/src/compression.js.map
│   │   │   ├── ./adblocker/dist/src/config.cjs
│   │   │   ├── ./adblocker/dist/src/config.cjs.map
│   │   │   ├── ./adblocker/dist/src/config.d.ts
│   │   │   ├── ./adblocker/dist/src/config.js
│   │   │   ├── ./adblocker/dist/src/config.js.map
│   │   │   ├── ./adblocker/dist/src/crc32.cjs
│   │   │   ├── ./adblocker/dist/src/crc32.cjs.map
│   │   │   ├── ./adblocker/dist/src/crc32.d.ts
│   │   │   ├── ./adblocker/dist/src/crc32.js
│   │   │   ├── ./adblocker/dist/src/crc32.js.map
│   │   │   ├── ./adblocker/dist/src/data-view.cjs
│   │   │   ├── ./adblocker/dist/src/data-view.cjs.map
│   │   │   ├── ./adblocker/dist/src/data-view.d.ts
│   │   │   ├── ./adblocker/dist/src/data-view.js
│   │   │   ├── ./adblocker/dist/src/data-view.js.map
│   │   │   ├── ./adblocker/dist/src/encoding.cjs
│   │   │   ├── ./adblocker/dist/src/encoding.cjs.map
│   │   │   ├── ./adblocker/dist/src/encoding.d.ts
│   │   │   ├── ./adblocker/dist/src/encoding.js
│   │   │   ├── ./adblocker/dist/src/encoding.js.map
│   │   │   ├── ./adblocker/dist/src/engine
│   │   │   ├── ./adblocker/dist/src/events.cjs
│   │   │   ├── ./adblocker/dist/src/events.cjs.map
│   │   │   ├── ./adblocker/dist/src/events.d.ts
│   │   │   ├── ./adblocker/dist/src/events.js
│   │   │   ├── ./adblocker/dist/src/events.js.map
│   │   │   ├── ./adblocker/dist/src/fetch.cjs
│   │   │   ├── ./adblocker/dist/src/fetch.cjs.map
│   │   │   ├── ./adblocker/dist/src/fetch.d.ts
│   │   │   ├── ./adblocker/dist/src/fetch.js
│   │   │   ├── ./adblocker/dist/src/fetch.js.map
│   │   │   ├── ./adblocker/dist/src/filters
│   │   │   ├── ./adblocker/dist/src/html-filtering.cjs
│   │   │   ├── ./adblocker/dist/src/html-filtering.cjs.map
│   │   │   ├── ./adblocker/dist/src/html-filtering.d.ts
│   │   │   ├── ./adblocker/dist/src/html-filtering.js
│   │   │   ├── ./adblocker/dist/src/html-filtering.js.map
│   │   │   ├── ./adblocker/dist/src/lists.cjs
│   │   │   ├── ./adblocker/dist/src/lists.cjs.map
│   │   │   ├── ./adblocker/dist/src/lists.d.ts
│   │   │   ├── ./adblocker/dist/src/lists.js
│   │   │   ├── ./adblocker/dist/src/lists.js.map
│   │   │   ├── ./adblocker/dist/src/preprocessor.cjs
│   │   │   ├── ./adblocker/dist/src/preprocessor.cjs.map
│   │   │   ├── ./adblocker/dist/src/preprocessor.d.ts
│   │   │   ├── ./adblocker/dist/src/preprocessor.js
│   │   │   ├── ./adblocker/dist/src/preprocessor.js.map
│   │   │   ├── ./adblocker/dist/src/punycode.cjs
│   │   │   ├── ./adblocker/dist/src/punycode.cjs.map
│   │   │   ├── ./adblocker/dist/src/punycode.d.ts
│   │   │   ├── ./adblocker/dist/src/punycode.js
│   │   │   ├── ./adblocker/dist/src/punycode.js.map
│   │   │   ├── ./adblocker/dist/src/queue-microtask.cjs
│   │   │   ├── ./adblocker/dist/src/queue-microtask.cjs.map
│   │   │   ├── ./adblocker/dist/src/queue-microtask.d.ts
│   │   │   ├── ./adblocker/dist/src/queue-microtask.js
│   │   │   ├── ./adblocker/dist/src/queue-microtask.js.map
│   │   │   ├── ./adblocker/dist/src/request.cjs
│   │   │   ├── ./adblocker/dist/src/request.cjs.map
│   │   │   ├── ./adblocker/dist/src/request.d.ts
│   │   │   ├── ./adblocker/dist/src/request.js
│   │   │   ├── ./adblocker/dist/src/request.js.map
│   │   │   ├── ./adblocker/dist/src/resources.cjs
│   │   │   ├── ./adblocker/dist/src/resources.cjs.map
│   │   │   ├── ./adblocker/dist/src/resources.d.ts
│   │   │   ├── ./adblocker/dist/src/resources.js
│   │   │   ├── ./adblocker/dist/src/resources.js.map
│   │   │   ├── ./adblocker/dist/src/tokens-buffer.cjs
│   │   │   ├── ./adblocker/dist/src/tokens-buffer.cjs.map
│   │   │   ├── ./adblocker/dist/src/tokens-buffer.d.ts
│   │   │   ├── ./adblocker/dist/src/tokens-buffer.js
│   │   │   ├── ./adblocker/dist/src/tokens-buffer.js.map
│   │   │   ├── ./adblocker/dist/src/utils.cjs
│   │   │   ├── ./adblocker/dist/src/utils.cjs.map
│   │   │   ├── ./adblocker/dist/src/utils.d.ts
│   │   │   ├── ./adblocker/dist/src/utils.js
│   │   │   └── ./adblocker/dist/src/utils.js.map
│   │   ├── ./adblocker/dist/test
│   │   │   ├── ./adblocker/dist/test/compact-set.test.d.ts
│   │   │   ├── ./adblocker/dist/test/compact-set.test.js
│   │   │   ├── ./adblocker/dist/test/data
│   │   │   ├── ./adblocker/dist/test/data-view.test.d.ts
│   │   │   ├── ./adblocker/dist/test/data-view.test.js
│   │   │   ├── ./adblocker/dist/test/engine
│   │   │   ├── ./adblocker/dist/test/fetch.test.d.ts
│   │   │   ├── ./adblocker/dist/test/fetch.test.js
│   │   │   ├── ./adblocker/dist/test/filters-size-estimation.test.d.ts
│   │   │   ├── ./adblocker/dist/test/filters-size-estimation.test.js
│   │   │   ├── ./adblocker/dist/test/html-filtering.test.d.ts
│   │   │   ├── ./adblocker/dist/test/html-filtering.test.js
│   │   │   ├── ./adblocker/dist/test/lists.test.d.ts
│   │   │   ├── ./adblocker/dist/test/lists.test.js
│   │   │   ├── ./adblocker/dist/test/matching.test.d.ts
│   │   │   ├── ./adblocker/dist/test/matching.test.js
│   │   │   ├── ./adblocker/dist/test/optimizer.test.d.ts
│   │   │   ├── ./adblocker/dist/test/optimizer.test.js
│   │   │   ├── ./adblocker/dist/test/parsing.test.d.ts
│   │   │   ├── ./adblocker/dist/test/parsing.test.js
│   │   │   ├── ./adblocker/dist/test/preprocessor.test.d.ts
│   │   │   ├── ./adblocker/dist/test/preprocessor.test.js
│   │   │   ├── ./adblocker/dist/test/punycode.test.d.ts
│   │   │   ├── ./adblocker/dist/test/punycode.test.js
│   │   │   ├── ./adblocker/dist/test/request.test.d.ts
│   │   │   ├── ./adblocker/dist/test/request.test.js
│   │   │   ├── ./adblocker/dist/test/resources.test.d.ts
│   │   │   ├── ./adblocker/dist/test/resources.test.js
│   │   │   ├── ./adblocker/dist/test/reverse-index.test.d.ts
│   │   │   ├── ./adblocker/dist/test/reverse-index.test.js
│   │   │   ├── ./adblocker/dist/test/serialization.test.d.ts
│   │   │   ├── ./adblocker/dist/test/serialization.test.js
│   │   │   ├── ./adblocker/dist/test/utils.d.ts
│   │   │   ├── ./adblocker/dist/test/utils.js
│   │   │   ├── ./adblocker/dist/test/utils.test.d.ts
│   │   │   └── ./adblocker/dist/test/utils.test.js
│   │   ├── ./adblocker/dist/tools
│   │   │   ├── ./adblocker/dist/tools/auto-bump-engine-version.d.ts
│   │   │   ├── ./adblocker/dist/tools/auto-bump-engine-version.js
│   │   │   ├── ./adblocker/dist/tools/bench-metadata.d.ts
│   │   │   ├── ./adblocker/dist/tools/bench-metadata.js
│   │   │   ├── ./adblocker/dist/tools/engine-size.d.ts
│   │   │   ├── ./adblocker/dist/tools/engine-size.js
│   │   │   ├── ./adblocker/dist/tools/generate_compression_codebooks.d.ts
│   │   │   ├── ./adblocker/dist/tools/generate_compression_codebooks.js
│   │   │   ├── ./adblocker/dist/tools/priorities.d.ts
│   │   │   ├── ./adblocker/dist/tools/priorities.js
│   │   │   ├── ./adblocker/dist/tools/stress-test-engine-update.d.ts
│   │   │   └── ./adblocker/dist/tools/stress-test-engine-update.js
│   │   ├── ./adblocker/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker/dist/types
│   │       ├── ./adblocker/dist/types/adblocker.d.cts
│   │       ├── ./adblocker/dist/types/adblocker.d.ts
│   │       ├── ./adblocker/dist/types/src
│   │       ├── ./adblocker/dist/types/test
│   │       └── ./adblocker/dist/types/tools
│   ├── ./adblocker-content/dist
│   │   ├── ./adblocker-content/dist/adblocker.cjs
│   │   ├── ./adblocker-content/dist/adblocker.cjs.map
│   │   ├── ./adblocker-content/dist/adblocker.d.ts
│   │   ├── ./adblocker-content/dist/adblocker.js
│   │   ├── ./adblocker-content/dist/adblocker.js.map
│   │   ├── ./adblocker-content/dist/adblocker.umd.min.js
│   │   ├── ./adblocker-content/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker-content/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-content/dist/types
│   │       ├── ./adblocker-content/dist/types/adblocker.d.cts
│   │       └── ./adblocker-content/dist/types/adblocker.d.ts
│   ├── ./adblocker-electron/dist
│   │   ├── ./adblocker-electron/dist/adblocker.cjs
│   │   ├── ./adblocker-electron/dist/adblocker.cjs.map
│   │   ├── ./adblocker-electron/dist/adblocker.d.ts
│   │   ├── ./adblocker-electron/dist/adblocker.js
│   │   ├── ./adblocker-electron/dist/adblocker.js.map
│   │   ├── ./adblocker-electron/dist/adblocker.umd.min.js
│   │   ├── ./adblocker-electron/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker-electron/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-electron/dist/types
│   │       ├── ./adblocker-electron/dist/types/adblocker.d.cts
│   │       └── ./adblocker-electron/dist/types/adblocker.d.ts
│   ├── ./adblocker-electron-preload/dist
│   │   ├── ./adblocker-electron-preload/dist/preload.cjs
│   │   ├── ./adblocker-electron-preload/dist/preload.cjs.map
│   │   ├── ./adblocker-electron-preload/dist/preload.d.ts
│   │   ├── ./adblocker-electron-preload/dist/preload.js
│   │   ├── ./adblocker-electron-preload/dist/preload.js.map
│   │   ├── ./adblocker-electron-preload/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-electron-preload/dist/types
│   │       ├── ./adblocker-electron-preload/dist/types/preload.d.cts
│   │       └── ./adblocker-electron-preload/dist/types/preload.d.ts
│   ├── ./adblocker-extended-selectors/dist
│   │   ├── ./adblocker-extended-selectors/dist/adblocker.cjs
│   │   ├── ./adblocker-extended-selectors/dist/adblocker.cjs.map
│   │   ├── ./adblocker-extended-selectors/dist/adblocker.d.ts
│   │   ├── ./adblocker-extended-selectors/dist/adblocker.js
│   │   ├── ./adblocker-extended-selectors/dist/adblocker.js.map
│   │   ├── ./adblocker-extended-selectors/dist/adblocker.umd.min.js
│   │   ├── ./adblocker-extended-selectors/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker-extended-selectors/dist/src
│   │   │   ├── ./adblocker-extended-selectors/dist/src/eval.cjs
│   │   │   ├── ./adblocker-extended-selectors/dist/src/eval.cjs.map
│   │   │   ├── ./adblocker-extended-selectors/dist/src/eval.d.ts
│   │   │   ├── ./adblocker-extended-selectors/dist/src/eval.js
│   │   │   ├── ./adblocker-extended-selectors/dist/src/eval.js.map
│   │   │   ├── ./adblocker-extended-selectors/dist/src/extended.cjs
│   │   │   ├── ./adblocker-extended-selectors/dist/src/extended.cjs.map
│   │   │   ├── ./adblocker-extended-selectors/dist/src/extended.d.ts
│   │   │   ├── ./adblocker-extended-selectors/dist/src/extended.js
│   │   │   ├── ./adblocker-extended-selectors/dist/src/extended.js.map
│   │   │   ├── ./adblocker-extended-selectors/dist/src/parse.cjs
│   │   │   ├── ./adblocker-extended-selectors/dist/src/parse.cjs.map
│   │   │   ├── ./adblocker-extended-selectors/dist/src/parse.d.ts
│   │   │   ├── ./adblocker-extended-selectors/dist/src/parse.js
│   │   │   ├── ./adblocker-extended-selectors/dist/src/parse.js.map
│   │   │   ├── ./adblocker-extended-selectors/dist/src/types.cjs
│   │   │   ├── ./adblocker-extended-selectors/dist/src/types.cjs.map
│   │   │   ├── ./adblocker-extended-selectors/dist/src/types.d.ts
│   │   │   ├── ./adblocker-extended-selectors/dist/src/types.js
│   │   │   └── ./adblocker-extended-selectors/dist/src/types.js.map
│   │   ├── ./adblocker-extended-selectors/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-extended-selectors/dist/types
│   │       ├── ./adblocker-extended-selectors/dist/types/adblocker.d.cts
│   │       ├── ./adblocker-extended-selectors/dist/types/adblocker.d.ts
│   │       └── ./adblocker-extended-selectors/dist/types/src
│   ├── ./adblocker-playwright/dist
│   │   ├── ./adblocker-playwright/dist/adblocker.cjs
│   │   ├── ./adblocker-playwright/dist/adblocker.cjs.map
│   │   ├── ./adblocker-playwright/dist/adblocker.d.ts
│   │   ├── ./adblocker-playwright/dist/adblocker.js
│   │   ├── ./adblocker-playwright/dist/adblocker.js.map
│   │   ├── ./adblocker-playwright/dist/adblocker.umd.min.js
│   │   ├── ./adblocker-playwright/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker-playwright/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-playwright/dist/types
│   │       ├── ./adblocker-playwright/dist/types/adblocker.d.cts
│   │       └── ./adblocker-playwright/dist/types/adblocker.d.ts
│   ├── ./adblocker-puppeteer/dist
│   │   ├── ./adblocker-puppeteer/dist/adblocker.cjs
│   │   ├── ./adblocker-puppeteer/dist/adblocker.cjs.map
│   │   ├── ./adblocker-puppeteer/dist/adblocker.d.ts
│   │   ├── ./adblocker-puppeteer/dist/adblocker.js
│   │   ├── ./adblocker-puppeteer/dist/adblocker.js.map
│   │   ├── ./adblocker-puppeteer/dist/adblocker.umd.min.js
│   │   ├── ./adblocker-puppeteer/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker-puppeteer/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-puppeteer/dist/types
│   │       ├── ./adblocker-puppeteer/dist/types/adblocker.d.cts
│   │       └── ./adblocker-puppeteer/dist/types/adblocker.d.ts
│   ├── ./adblocker-webextension/dist
│   │   ├── ./adblocker-webextension/dist/adblocker.cjs
│   │   ├── ./adblocker-webextension/dist/adblocker.cjs.map
│   │   ├── ./adblocker-webextension/dist/adblocker.d.ts
│   │   ├── ./adblocker-webextension/dist/adblocker.js
│   │   ├── ./adblocker-webextension/dist/adblocker.js.map
│   │   ├── ./adblocker-webextension/dist/adblocker.umd.min.js
│   │   ├── ./adblocker-webextension/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker-webextension/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-webextension/dist/types
│   │       ├── ./adblocker-webextension/dist/types/adblocker.d.cts
│   │       └── ./adblocker-webextension/dist/types/adblocker.d.ts
│   ├── ./adblocker-webextension-cosmetics/dist
│   │   ├── ./adblocker-webextension-cosmetics/dist/adblocker.cjs
│   │   ├── ./adblocker-webextension-cosmetics/dist/adblocker.cjs.map
│   │   ├── ./adblocker-webextension-cosmetics/dist/adblocker.d.ts
│   │   ├── ./adblocker-webextension-cosmetics/dist/adblocker.js
│   │   ├── ./adblocker-webextension-cosmetics/dist/adblocker.js.map
│   │   ├── ./adblocker-webextension-cosmetics/dist/adblocker.umd.min.js
│   │   ├── ./adblocker-webextension-cosmetics/dist/adblocker.umd.min.js.map
│   │   ├── ./adblocker-webextension-cosmetics/dist/tsconfig.tsbuildinfo
│   │   └── ./adblocker-webextension-cosmetics/dist/types
│   │       ├── ./adblocker-webextension-cosmetics/dist/types/adblocker.d.cts
│   │       └── ./adblocker-webextension-cosmetics/dist/types/adblocker.d.ts
    ├── ./adblocker-webextension-example/dist
    │   ├── ./adblocker-webextension-example/dist/background.iife.js
    │   ├── ./adblocker-webextension-example/dist/background.iife.js.map
    │   ├── ./adblocker-webextension-example/dist/content-script.iife.js
    │   ├── ./adblocker-webextension-example/dist/content-script.iife.js.map
    │   ├── ./adblocker-webextension-example/dist/src
    │   │   ├── ./adblocker-webextension-example/dist/src/background.js
    │   │   ├── ./adblocker-webextension-example/dist/src/content-script.js
    │   │   └── ./adblocker-webextension-example/dist/src/tsconfig.tsbuildinfo
    │   └── ./adblocker-webextension-example/dist/types
    │       ├── ./adblocker-webextension-example/dist/types/background.d.ts
    │       └── ./adblocker-webextension-example/dist/types/content-script.d.ts
```

</details>

**Comment by seia-soto** [comment]
Please, note that examples are not updated in this PR.

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ dist
 node_modules
 .nyc_output
 .DS_Store
+tsconfig.tsbuildinfo
```
Comment (chrmod) [inline .gitignore:6]: I not think this or `.rollup.cache/` need to be here. Have noticed them created when the rollup config is not exactly as needed, but once finished, those were not created any longer.

**Review by chrmod — APPROVED** [review]

**Pull request merged**